Local imports crash with AWS SDK error when importing from local filesystem

**Describe the bug**

```
docker-compose up -d
bloop run application -- import-catalog --catalog-root ~/src/stacframes/examples/aviris/catalog/catalog.jso
n --db-host localhost --db-password franklin --db-name franklin
```

Throws:
```
[E] Exception in thread "main" java.lang.ExceptionInInitializerError                                                                      
[E]     at com.azavea.franklin.crawler.CatalogStacImport.readCollection(CatalogStacImport.scala:60)                                       
[E]     at com.azavea.franklin.crawler.CatalogStacImport.readRoot(CatalogStacImport.scala:73)                                             
[E]     at com.azavea.franklin.crawler.CatalogStacImport.runIO(CatalogStacImport.scala:255)                                               
[E]     at com.azavea.franklin.api.commands.Commands$.runCatalogImport(Commands.scala:94)                                                 
[E]     at com.azavea.franklin.api.Server$.$anonfun$run$1(Server.scala:140)                                                               
[E]     at scala.util.Either.map(Either.scala:353)                                                                                        
[E]     at com.azavea.franklin.api.Server$.run(Server.scala:130)                                                                          
[E]     at cats.effect.IOApp.$anonfun$main$3(IOApp.scala:68)                                                                              
[E]     at cats.effect.internals.IOAppPlatform$.mainFiber(IOAppPlatform.scala:40)                                                         
[E]     at cats.effect.internals.IOAppPlatform$.main(IOAppPlatform.scala:25)                                                              
[E]     at cats.effect.IOApp.main(IOApp.scala:68)                                                                                         
[E]     at cats.effect.IOApp.main$(IOApp.scala:67)                                                                                        
[E]     at com.azavea.franklin.api.Server$.main(Server.scala:34)                                                                          
[E]     at com.azavea.franklin.api.Server.main(Server.scala)                                                                              
[E] Caused by: com.amazonaws.SdkClientException: Unable to find a region via the region provider chain. Must provide an explicit region in
 the builder or setup environment to supply a region.
```

If I set `AWS_REGION` in my shell, e.g. `export AWS_REGION=us-east-1` then rerun the import command, imports proceed normally.

**Expected behavior**

Even if my AWS environment isn't quite configured correctly (no default region set via `~/.aws/config`), I wouldn't expect the AWS SDK to be loaded and my credentials read when not performing an import from AWS services.

Local imports should run successfully without loading the AWS SDK.

--- a/application/src/main/scala/com/azavea/franklin/api/Server.scala
+++ b/application/src/main/scala/com/azavea/franklin/api/Server.scala
@@ -22,6 +22,7 @@ import org.http4s.implicits._
 import org.http4s.server.blaze._
 import org.http4s.server.middleware._
 import org.http4s.server.{Router, Server => HTTP4sServer}
+import sttp.client.asynchttpclient.cats.AsyncHttpClientCatsBackend
 import sttp.tapir.docs.openapi._
 import sttp.tapir.openapi.circe.yaml._
 import sttp.tapir.swagger.http4s.SwaggerHttp4s
@@ -139,16 +140,20 @@ $$$$
           createServer(apiConfig, dbConfig).use(_ => IO.never).as(ExitCode.Success)
       case RunMigrations(config) => runMigrations(config)
       case RunCatalogImport(catalogRoot, dbConfig, dryRun) =>
-        runCatalogImport(catalogRoot, dbConfig, dryRun) map { _ => ExitCode.Success }
+        AsyncHttpClientCatsBackend[IO]() flatMap { implicit backend =>
+          runCatalogImport(catalogRoot, dbConfig, dryRun) map { _ => ExitCode.Success }
+        }
       case RunItemsImport(collectionId, itemUris, dbConfig, dryRun) => {
-        runStacItemImport(collectionId, itemUris, dbConfig, dryRun) map {
-          case Left(error) => {
-            println(s"Import failed: $error")
-            ExitCode.Error
-          }
-          case Right(numItemsImported) => {
-            println(s"Import succesful: ${numItemsImported} items imported")
-            ExitCode.Success
+        AsyncHttpClientCatsBackend[IO]() flatMap { implicit backend =>
+          runStacItemImport(collectionId, itemUris, dbConfig, dryRun) map {
+            case Left(error) => {
+              println(s"Import failed: $error")
+              ExitCode.Error
+            }
+            case Right(numItemsImported) => {
+              println(s"Import succesful: ${numItemsImported} items imported")
+              ExitCode.Success
+            }
           }
         }
       }

--- a/application/src/main/scala/com/azavea/franklin/commands/Commands.scala
+++ b/application/src/main/scala/com/azavea/franklin/commands/Commands.scala
@@ -11,6 +11,7 @@ import doobie.Transactor
 import doobie.free.connection.{rollback, setAutoCommit, unit}
 import doobie.util.transactor.Strategy
 import org.flywaydb.core.Flyway
+import sttp.client.{NothingT, SttpBackend}
 
 object Commands {
 
@@ -78,7 +79,7 @@ object Commands {
       itemUris: NonEmptyList[String],
       config: DatabaseConfig,
       dryRun: Boolean
-  )(implicit cs: ContextShift[IO]) = {
+  )(implicit cs: ContextShift[IO], backend: SttpBackend[IO, Nothing, NothingT]) = {
     val xa = config.getTransactor(dryRun)
     new StacItemImporter(collectionId, itemUris).runIO(xa)
   }
@@ -88,7 +89,8 @@ object Commands {
       config: DatabaseConfig,
       dryRun: Boolean
   )(
-      implicit contextShift: ContextShift[IO]
+      implicit contextShift: ContextShift[IO],
+      backend: SttpBackend[IO, Nothing, NothingT]
   ): IO[Unit] = {
     val xa = config.getTransactor(dryRun)
     new CatalogStacImport(stacCatalog).runIO(xa)

--- a/application/src/main/scala/com/azavea/franklin/crawler/CatalogStacImport.scala
+++ b/application/src/main/scala/com/azavea/franklin/crawler/CatalogStacImport.scala
@@ -22,6 +22,7 @@ import io.circe.Decoder
 import io.circe.JsonObject
 import io.circe.parser.decode
 import io.circe.syntax._
+import sttp.client.{NothingT, SttpBackend}
 
 import scala.concurrent.ExecutionContext.Implicits.global
 
@@ -56,7 +57,9 @@ class CatalogStacImport(val catalogRoot: String) {
     )
   }
 
-  def readCollection(path: String, xa: Transactor[IO]): IO[StacCollection] =
+  def readCollection(path: String, xa: Transactor[IO])(
+      implicit backend: SttpBackend[IO, Nothing, NothingT]
+  ): IO[StacCollection] =
     readJsonFromPath[StacCollection](path) flatMap { collection =>
       StacCollectionDao
         .insertStacCollection(
@@ -69,7 +72,9 @@ class CatalogStacImport(val catalogRoot: String) {
         .transact(xa) map { _ => collection }
     }
 
-  def readRoot(xa: Transactor[IO]): IO[String] =
+  def readRoot(
+      xa: Transactor[IO]
+  )(implicit backend: SttpBackend[IO, Nothing, NothingT]): IO[String] =
     (readCollection(catalogRoot, xa) map { _.id }) orElse (readJsonFromPath[StacCatalog](
       catalogRoot
     ) map { _.id })
@@ -77,6 +82,8 @@ class CatalogStacImport(val catalogRoot: String) {
   private def createCollectionForGeoJsonAsset(
       forItem: StacItem,
       fromPath: String
+  )(
+      implicit backend: SttpBackend[IO, Nothing, NothingT]
   ): IO[List[(Map[String, StacItemAsset], CollectionWrapper)]] = {
     val geojsonAssets = forItem.assets.toList.filter {
       case (_, asset) => asset._type === Some(`application/geo+json`)
@@ -193,7 +200,9 @@ class CatalogStacImport(val catalogRoot: String) {
     } yield colInsert
   }
 
-  def allChildPaths(xa: Transactor[IO]): StateT[IO, String, List[String]] =
+  def allChildPaths(
+      xa: Transactor[IO]
+  )(implicit backend: SttpBackend[IO, Nothing, NothingT]): StateT[IO, String, List[String]] =
     StateT[IO, String, List[String]] { (root: String) =>
       for {
         rootLinks <- readCollection(root, xa) map { _.links } orElse {
@@ -212,7 +221,9 @@ class CatalogStacImport(val catalogRoot: String) {
       } yield childHrefs
     }
 
-  def insertItemsForAbsHref(absHref: String, xa: Transactor[IO]): IO[Unit] = {
+  def insertItemsForAbsHref(absHref: String, xa: Transactor[IO])(
+      implicit backend: SttpBackend[IO, Nothing, NothingT]
+  ): IO[Unit] = {
     ((readCollection(absHref, xa) map { collection =>
       (collection.links, Option(collection.id))
     } orElse (readJsonFromPath[
@@ -250,7 +261,7 @@ class CatalogStacImport(val catalogRoot: String) {
     }).void
   }
 
-  def runIO(xa: Transactor[IO]): IO[Unit] =
+  def runIO(xa: Transactor[IO])(implicit backend: SttpBackend[IO, Nothing, NothingT]): IO[Unit] =
     for {
       _             <- readRoot(xa)
       allChildHrefs <- allChildPaths(xa).runA(catalogRoot)

--- a/application/src/main/scala/com/azavea/franklin/crawler/StacIO.scala
+++ b/application/src/main/scala/com/azavea/franklin/crawler/StacIO.scala
@@ -65,7 +65,7 @@ object StacIO {
 
   def readJsonFromPath[T: Decoder](
       path: String
-  )(implicit contextShift: ContextShift[IO], logger: Logger[IO]): IO[T] = {
+  )(implicit logger: Logger[IO], backend: SttpBackend[IO, Nothing, NothingT]): IO[T] = {
     if (path.startsWith("http")) {
       {
         readJsonFromHttp(path)
@@ -84,19 +84,20 @@ object StacIO {
 
   def readJsonFromHttp[T: Decoder](
       url: String
-  )(implicit contextShift: ContextShift[IO], logger: Logger[IO]): IO[T] =
-    AsyncHttpClientCatsBackend[IO]().flatMap { implicit backend =>
-      for {
-        response <- basicRequest.get(uri"$url").response(asJson[T]).send[IO]
-        decoded <- response.body match {
-          case Left(e @ DeserializationError(_, err)) =>
-            logCirceError(s"$url", err) *> IO.raiseError(e)
-          case Left(e @ HttpError(_, code)) =>
-            logBadResponse(s"$url", code.code) *> IO.raiseError(e)
-          case Right(body) => IO.pure(body)
-        }
-      } yield decoded
-    }
+  )(
+      implicit logger: Logger[IO],
+      backend: SttpBackend[IO, Nothing, NothingT]
+  ): IO[T] =
+    for {
+      response <- basicRequest.get(uri"$url").response(asJson[T]).send[IO]
+      decoded <- response.body match {
+        case Left(e @ DeserializationError(_, err)) =>
+          logCirceError(s"$url", err) *> IO.raiseError(e)
+        case Left(e @ HttpError(_, code)) =>
+          logBadResponse(s"$url", code.code) *> IO.raiseError(e)
+        case Right(body) => IO.pure(body)
+      }
+    } yield decoded
 
   private def getPrefix(absPath: String): String = absPath.split("/").dropRight(1).mkString("/")
 
@@ -120,7 +121,11 @@ object StacIO {
       path: String,
       rewriteSourceIfPresent: Boolean,
       inCollectionId: Option[String]
-  )(implicit contextShift: ContextShift[IO], logger: Logger[IO]): IO[StacItem] = {
+  )(
+      implicit contextShift: ContextShift[IO],
+      logger: Logger[IO],
+      backend: SttpBackend[IO, Nothing, NothingT]
+  ): IO[StacItem] = {
     val readIO = readJsonFromPath[StacItem](path)
     if (!rewriteSourceIfPresent) (logger.debug(s"Not rewriting source link at $path") *> readIO)
     else {

--- a/application/src/main/scala/com/azavea/franklin/crawler/StacIO.scala
+++ b/application/src/main/scala/com/azavea/franklin/crawler/StacIO.scala
@@ -37,7 +37,7 @@ object StacIO {
   private def logBadResponse(path: String, code: Int)(implicit logger: Logger[IO]): IO[Unit] =
     logger.error(s"The server responsible for $path rejected my request with a status of $code")
 
-  val s3 = AmazonS3ClientBuilder
+  lazy val s3 = AmazonS3ClientBuilder
     .standard()
     .withForceGlobalBucketAccessEnabled(true)
     .build()

--- a/application/src/main/scala/com/azavea/franklin/crawler/StacItemImporter.scala
+++ b/application/src/main/scala/com/azavea/franklin/crawler/StacItemImporter.scala
@@ -10,9 +10,11 @@ import com.azavea.stac4s._
 import doobie.implicits._
 import doobie.util.transactor.Transactor
 import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import sttp.client.{NothingT, SttpBackend}
 
 class StacItemImporter(val collectionId: String, val itemUris: NonEmptyList[String])(
-    implicit contextShift: ContextShift[IO]
+    implicit contextShift: ContextShift[IO],
+    backend: SttpBackend[IO, Nothing, NothingT]
 ) {
 
   implicit def logger = Slf4jLogger.getLogger[IO]

